marketplace/optimizely/test/AppPage.spec.js

```diff
@@ -52,13 +38,39 @@ describe('AppPage', () => {
 
     const sdk = { ...basicProps.sdk, platformAlpha, space };
 
-    const props = { ...basicProps, accessToken: '123', client, sdk };
+    const props = { ...basicProps, accessToken: '123', sdk };
 
-    const { container } = render(<AppPage {...props} />);
+    const { container, getByTestId } = render(<AppPage {...props} />);
+
+    await wait(() => {
+      if (!configFunc) {
+        throw '';
+      }
+    });
 
-    await wait();
     await configFunc();
-    await wait();
+    expect(container).toMatchSnapshot();
+
+    act(() => {
+      fireEvent.click(getByTestId('add-content'));
+    });
+
+    getByTestId('content-type-selector').click();
+
+    getByTestId('content-type-selector').firstChild.click();
+    expect(getByTestId('content-type-selector')).toMatchSnapshot();
+  });
+
+  it('should render the AppPage with Features and connect button', () => {
+    const { container } = render(<AppPage {...basicProps} />);
+
+    expect(container).toMatchSnapshot();
```

Comment:
I'm not a biggest fan of making snapshots for the whole components. As for me it's making more noise during development rather than giving a useful information. I would query important elements and match their content using `toMatchInlineSnapshot`

Reply:
I see where you're coming from but this snapshot feels important as the whole picture. The reason is this specific instance of the `AppPage`, has two main elements: the Features list and the Optimizely oAuth button. By taking this snapshot I was able to verify both exist while other elements do not. A diff in this snapshot later on could indicate more elements are showing that potentially shouldn't be.

If I were to individually select the elements I am expecting, then I'm also ignoring any other elements that may or may not show up in the snapshot, which is exactly what I don't want to do.

Reply:
Well you can select several parts in one test to verify that all elements exist in a snapshot. But I don't insist ;)

Reply:
I'm going to keep it the way it is because I don't want to select several parts. That is inclusive. I want to be exclusive and have the whole snapshot signify what should be there and what shouldn't be.